build.sh

```diff
@@ -1,20 +1,28 @@
 #/usr/bin/env bash
 
-ios_version="16.1"
+set -e
+
+ios_version=$(xcrun --sdk iphoneos --show-sdk-platform-version)
+macosx_version=$(xcrun --sdk macosx --show-sdk-platform-version)
 
 targets=(
   "arm64-apple-ios$ios_version"
   "arm64-apple-ios$ios_version-simulator"
-  # "x86_64-apple-ios$ios_version-simulator"
+  "x86_64-apple-ios$ios_version-simulator"
+  "arm64-apple-darwin$macosx_version"
+  "x86_64-apple-darwin$macosx_version"
 )
 
 sdk_names=(
   "iphoneos$ios_version"
   "iphonesimulator$ios_version"
-  # "iphonesimulator$ios_version"
+  "iphonesimulator$ios_version"
+  "macosx$macosx_version"
+  "macosx$macosx_version"
```

Comment:
Because when accessing it we use the same indexing as for targets array

Reply:
What do you think about using [jq](https://stedolan.github.io/jq/) and redoing this to json? I think it could clean up some code in here (also allow to get rid of printf workaround from the loop below), but on the other hand it is an external dependency that has to be installed.


Alternatively I think bash associative arrays would also play nicely in here, more or less like so:
```
$ declare -A targets=( ["arm64-apple-ios$ios_version"]="iphoneos$ios_version" ["arm64-apple-ios$ios_version-simulator"]="iphonesimulator$ios_version" )

$ for key in "${!targets[@]}"; do echo "$key --- ${targets[$key]}"; done
> arm64-apple-ios16.0-simulator --- iphonesimulator16.0
> arm64-apple-ios16.0 --- iphoneos16.0
```

although from what I know macos ships with some very old bash version by default (3.2 if I'm not mistaken), and associative arrays were added in bash 4, so you'd have to upgrade using brew. 

What do you think about these ideas?

Reply:
Sure, just one last request from me then - can you add comments in the code, stating why it is done the way it is done, to avoid future confusion? (in this case the order is very important, so someone changing the code should know that)

Reply:
Sure thing